woocommerce/woocommerce-ios
# Store creation: send a Tracks event with profiler answers when reaching store summary screen

Closes: #8644 


## Description


### Why

Source: p1673363982459279-slack-C045CUK1Y3U
Ref: pe5sF9-I3-p2#milestone-4-onboarding

> Since we have the profiling questions ready, but we are still waiting for the API endpoints, Anitaa get the approval from Samrat to release the profiling questions without the endpoint, collecting the answers somewhere: ideally on Tracks, unless you have a better tool in mind for that.

### Event

Event name: `*site_creation_profiler_data`
Event properties:
```
industry_group : automotive , food_drink , digital_products, etc
industry : fuel_dispensers , books, grocery_stores, long etc
user_commerce_journey : im_just_starting_my_business , im_already_selling_but_not_online, im_already_selling_online
ecommerce_platforms: amazon , big-cartel shopify , big-commerce , etc.
country: country ISO code
```

### Implementation

Since the API request was originally designed to fire after completing each profiler question, some changes were made to pass the answer to the later screens until it reaches the store summary screen where the new Tracks event is sent with all the answers from the 3 profiler questions. 2 new answer structs were created to encapsulate all the data needed to show on the store summary screen and send to the Tracks event (and later the API).

## Testing instructions


- Log in if needed
- Go to the Menu tab, and tap `Switch store`
- On the store picker, tap `+ Add a store`
- Tap `Create a new store` --> the domain selector should be shown with a placeholder image
- Enter some store name and tap `Continue`
- Select a category and tap `Continue`
- Select `I'm already selling online` and tap `Continue`
- Select multiple platforms and tap `Continue`
- Select a country and tap `Continue`
- Select or search for a domain, then tap `Continue` --> the store name, category, and country from the previous questions should be shown on the store summary screen. Also, a Tracks event should be tracked like: `🔵 Tracked site_creation_profiler_data, properties: [AnyHashable("country_code"): "AQ", AnyHashable("ecommerce_platforms"): "amazon,etsy,facebook-marketplace,google-shopping", AnyHashable("user_commerce_journey"): "im_already_selling_online", AnyHashable("industry"): "shoes", AnyHashable("industry_group"): "retail"]`
- Go back to the category question
- Tap `Skip` to skip the category question
- Tap `Skip` to skip the selling status question
- Tap `Continue` to continue with the default country code
- Select or search for a domain, then tap `Continue` --> the store name, category, and country from the previous questions should be shown on the store summary screen. Also, a Tracks event should be tracked like: `🔵 Tracked site_creation_profiler_data, properties: [AnyHashable("country_code"): "US"]`

## Screenshots


<img src="https://user-images.githubusercontent.com/1945542/212794922-c68fe25b-5b7e-4a78-9786-b0f4e32da8ce.png" width="300" />


---
